llama-index-integrations/vector_stores/llama-index-vector-stores-vectorx/llama_index/vector_stores/vectorx/base.py

```
@@ -0,0 +1,426 @@
+                ):
+                    op_symbol = reverse_operator_map.get(filter_item.operator)
+                    if not op_symbol:
+                        raise ValueError(
+                            f"Unsupported filter operator: {filter_item.operator}"
+                        )
+
+                    if filter_item.key not in filters:
+                        filters[filter_item.key] = {}
+
+                    filters[filter_item.key][op_symbol] = filter_item.value
+
+                # Case 2: Raw dict, e.g. {"category": {"$eq": "programming"}}
+                elif isinstance(filter_item, dict):
+                    for key, op_dict in filter_item.items():
+                        if isinstance(op_dict, dict):
+                            for op, val in op_dict.items():
+                                if key not in filters:
+                                    filters[key] = {}
+                                filters[key][op] = val
+                else:
+                    raise ValueError(f"Unsupported filter format: {filter_item}")
+
+        _logger.info(f"Final structured filters: {filters}")
+
+        # Use the query embedding if provided
+        if query.query_embedding is not None:
+            query_embedding = cast(List[float], query.query_embedding)
+            if query.alpha is not None and query.mode == VectorStoreQueryMode.HYBRID:
+                # Apply alpha scaling in hybrid mode
+                query_embedding = [v * query.alpha for v in query_embedding]
+
+        # Execute query
+        try:
+            results = self._vectorx_index.query(
+                vector=query_embedding,
+                top_k=query.similarity_top_k,
+                filter=filters if filters else None,
+                include_vectors=True,
+            )
+        except Exception as e:
+            _logger.error(f"Error querying VectorX: {e}")
+            return VectorStoreQueryResult(nodes=[], similarities=[], ids=[])
```

Not super sure returning an empty query and logging the error is  the best here: I would simply throw the error